calabash/Classes/FranklyServer/Routes/LPSuspendAppRoute.m

```diff
@@ -0,0 +1,97 @@
+#if ! __has_feature(objc_arc)
+#warning This file must be compiled with ARC. Use -fobjc-arc flag (or convert project to ARC).
+#endif
+
+#import <UIKit/UIKit.h>
+#import "LPSuspendAppRoute.h"
+#import "LPCocoaLumberjack.h"
+#import "LPInfoPlist.h"
+#import "LPInvoker.h"
+
+@interface UIApplication (LP_SUSPEND_APP_CATEGORY)
+
+- (void) suspend;
+
+@end
+
+@interface LPSuspendAppRoute ()
+
+- (CGFloat) durationWithDictionary:(NSDictionary *) arguments;
+
+@end
+
+@implementation LPSuspendAppRoute
+
+- (BOOL) supportsMethod:(NSString *) method atPath:(NSString *) path {
+  return [method isEqualToString:@"GET"];
+}
+
+- (CGFloat) durationWithDictionary:(NSDictionary *) arguments {
+  NSNumber *durationNumber = [arguments objectForKey:@"duration"];
+  if (durationNumber) {
+    return [durationNumber doubleValue];
+  } else {
+    return 2.0;
+  }
+}
+
+- (NSDictionary *) JSONResponseForMethod:(NSString *) method
+                                     URI:(NSString *) path
+                                     data:(NSDictionary *) data {
+
+  CGFloat duration = [self durationWithDictionary:data];
+
+
+  UIBackgroundTaskIdentifier __block task;
+
+  // I am not sure why this is necessary.  It does not appear to be called.
+  // I don't see the log message and I tried to force an app exit by inserting
+  // an `abort()` call.
+  task = [[UIApplication sharedApplication] beginBackgroundTaskWithName:@"Resume"
```

Comment:
@jmoody I would ask the same question - It appears you could accomplish the backgrounding by 
`[[UIApplication sharedApplication] suspend];` + `[LSApplicationWorkspace openApplicationWithBundleID:]` after a delay. It's a bit opaque me me how starting and ending a background task would help in this. Can you illuminate for me?